src/bundles/ipfs-provider.js

```diff
@@ -0,0 +1,323 @@
+}
+
+const asHttpClientOptions = (value) =>
+  typeof value === 'string' ? parseHTTPClientOptions(value) : readHTTPClinetOptions(value)
+
+const parseHTTPClientOptions = (input) => {
+  // Try parsing and reading as json
+  try {
+    return readHTTPClinetOptions(JSON.parse(input))
+  } catch (_) {}
+
+  // turn URL with inlined basic auth into client options object
+  try {
+    const uri = new URL(input)
+    const { username, password } = uri
+    if (username && password) {
+      return {
+        host: uri.hostname,
+        port: uri.port || (uri.protocol === 'https:' ? '443' : '80'),
+        protocol: uri.protocol.slice(0, -1), // trim out ':' at the end
+        apiPath: (uri.pathname !== '/' ? uri.pathname : 'api/v0'),
+        headers: {
+          authorization: `Basic ${btoa(username + ':' + password)}`
+        }
+      }
+    }
+  } catch (_) { }
+
+  return null
+}
+
+const readHTTPClinetOptions = (value) => {
+  // https://github.com/ipfs/js-ipfs/tree/master/packages/ipfs-http-client#importing-the-module-and-usage
+  if (value && (value.host || value.apiPath || value.protocol || value.port || value.headers)) {
+    return value
+  } else {
+    return null
+  }
+}
+
+const readSetting = (id) => {
+  /** @type {string|null} */
+  let setting = null
+  if (window.localStorage) {
+    try {
+      setting = window.localStorage.getItem(id)
+    } catch (error) {
+      console.error(`Error reading '${id}' value from localStorage`, error)
+    }
+
+    try {
+      return JSON.parse(setting || '')
+    } catch (_) {
+      // res was probably a string, so pass it on.
+      return setting
+    }
+  }
+
+  return setting
+}
+
+const writeSetting = (id, value) => {
+  try {
+    window.localStorage.setItem(id, JSON.stringify(value))
+  } catch (error) {
+    console.error(`Error writing '${id}' value to localStorage`, error)
+  }
+}
+
+let ipfs = null
+
+const extra = {
+  getIpfs () {
+    return ipfs
+  }
+}
+
+const selectors = {
+  selectIpfsReady: state => state.ipfs.ready,
+  selectIpfsProvider: state => state.ipfs.provider,
+  selectIpfsApiAddress: state => state.ipfs.apiAddress,
+  selectIpfsInvalidAddress: state => state.ipfs.invalidAddress,
+  selectIpfsInitFailed: state => state.ipfs.failed,
+  selectIpfsPendingFirstConnection: state => state.ipfs.pendingFirstConnection
+}
+
+const actions = {
+  doTryInitIpfs: () => async ({ store }) => {
+    // There is a code in `bundles/retry-init.js` that reacts to `IPFS_INIT`
+    // action and attempts to retry.
+    try {
+      await store.doInitIpfs()
+      return true
+    } catch (_) {
+      // Catches connection errors like timeouts
+      return false
+    }
+  },
+
+  doInitIpfs: () => perform('IPFS_INIT',
+    async (context) => {
+      const { apiAddress } = context.getState().ipfs
+
+      const result = await getIpfs({
+        loadHttpClientModule: () => HttpClient,
+        providers: [
+          providers.httpClient({ apiAddress })
+        ]
+      })
```

Comment:
We need to move `ipfs` from devDependencies to regular deps and switch ipfs-provider to js-ipfs backend:
https://github.com/ipfs-shipyard/ipfs-provider#jsipfs

(The sooner we switch to js-ipfs, the easier it will be for us to evaluate performance of the app in production)